Introduce the `GUILD_MEDIA` channel type (#6232)
hammerandchisel/discord-api-docs: docs/topics/Threads.md

> info
> More information about forum channels and how they appear in Discord can be found in the [Forum Channels FAQ](https://support.discord.com/hc/en-us/articles/6208479917079-Forum-Channels-FAQ#h_01G69FJQWTWN88HFEHK7Z6X79N)

## Media Channels

> A `GUILD_MEDIA` (type `16`) channel is similar to a `GUILD_FORUM` channel in that only threads can be created in them. Unless otherwise noted, threads in media channels behave in the same way as in forum channel - meaning they use the same endpoints and receive the same Gateway events. More information about media channels and how they appear in Discord can be found in the [media channels Help Center Article](https://creator-support.discord.com/hc/en-us/articles/14346342766743)

Reviewer: ```suggestion
A `GUILD_MEDIA` (type `16`) channel is similar to a `GUILD_FORUM` channel in that only threads can be created in them. Unless otherwise noted, threads in media channels behave in the same way as in forum channels - meaning they use the same endpoints and receive the same Gateway events. More information about media channels and how they appear in Discord can be found in the [Help Center Article](https://creator-support.discord.com/hc/en-us/articles/14346342766743).
```

> warn
> `GUILD_MEDIA` channels are in beta and still being actively developed. The API and other technical details are subject to change.

### Creating Threads in Forum and Media Channels

> Within thread-only channels, threads appear as posts. Threads can be created using the [`POST /channels/<channel_id>/threads`](#DOCS_RESOURCES_CHANNEL/start-thread-in-forum-channel) endpoint with [slightly different parameters](#DOCS_RESOURCES_CHANNEL/start-thread-in-forum-channel-jsonform-params) than threads in text channels. For example, when creating threads in a threads-only channel, a message is created that has the same ID as the thread which requires you to pass parameters for both a thread *and* a message.

Reviewer: I think it's not ideal to use the term "thread-only channel" without really explaining that it's referring to Forum and Media Channels. This might add add unnecessary confusion, especially given that in other places this is not abbreviated in this way.
Maybe the Forum and Media channel sections could be grouped under a "thread-only channels" section to make the definition more clear.

> Threads in thread-only channels have the same permissions behavior as threads in a text channel, inheriting all permissions from the parent channel, with one exception: creating a thread in a threads-only channel only requires the `SEND_MESSAGES` permission.

Reviewer: ```suggestion
Threads in thread-only channels have the same permissions behavior as threads in a text channel, inheriting all permissions from the parent channel, with one exception: creating a thread in a thread-only channel only requires the `SEND_MESSAGES` permission.
```

### Forum and Media Channel Fields

It's worth calling out a few details about fields specific to thread-only channels that may be important to keep in mind:

- The `last_message_id` field is the ID of the most recently created thread in that channel. As with messages, your app will not receive a `CHANNEL_UPDATE` event when the field is changed. Instead clients should update the value when receiving [Thread Create](#DOCS_TOPICS_GATEWAY_EVENTS/thread-create) events.
- The `topic` field is what is shown in the "Guidelines" section within the Discord client.

> - The `rate_limit_per_user` field limits how frequently threads can be created. There is a new `default_thread_rate_limit_per_user` field on thread-only channels as well, which limits how often messages can be sent _in a thread_. This field is copied into `rate_limit_per_user` on the thread at creation time.

Reviewer 2: The create guild channel endpoint respects `default_thread_rate_limit_per_user` for media channels (as well as announcement channels and forums channels). Perhaps this should be documented in the JSON parameters of this endpoint and the channel types should be noted in the channel structure.

Reviewer 3: * #5978

- The `available_tags` field can be set when creating or updating a channel, which determines which tags can be set on individual threads within the thread's `applied_tags` field.
- The `flags` field indicates any [channel flags](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) set for a thread-only channel. Currently only `REQUIRE_TAG` can be used, which requires that a tag from `available_tags` be specified when creating a thread in that channel.

All fields for channels, including thread-only channels, can be found in the [Channel Object](#DOCS_RESOURCES_CHANNEL/channel-object).

### Forum and Media Channel Thread Fields

> A thread can be pinned within a thread-only, which will be represented as the [`PINNED` flag](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) in the `flags` field within a thread-only channel. A thread that is pinned will have the `(1 << 1)` flag set, and archiving that thread will unset the flag. A pinned thread will *not* auto-archive.

Reviewer 3: this should say "thread-only channel"

> info
> More information about forum channels and how they appear in Discord can be found in the [Forum Channels FAQ](https://support.discord.com/hc/en-us/articles/6208479917079-Forum-Channels-FAQ#h_01G69FJQWTWN88HFEHK7Z6X79N)

## Media Channels

> A `GUILD_MEDIA` (type `16`) channel is similar to a `GUILD_FORUM` channel. Similar to forum channel, only threads can be created in them. Unless otherwise noted, threads in media channels behave in the same way as in forum channel - meaning they use the same endpoints and receive the same Gateway events.

Reviewer 4: ```suggestion
A `GUILD_MEDIA` (type `16`) channel is similar to a `GUILD_FORUM` channel in that only threads can be created in them. Unless otherwise noted, threads in media channels behave in the same way as in forum channel - meaning they use the same endpoints and receive the same Gateway events. More information about media channels and how they appear in Discord can be found in the [media channels Help Center Article](https://creator-support.discord.com/hc/en-us/articles/14346342766743)
```

> **This feature is still in active development. Many aspects are subject to change**

Reviewer 4: ```suggestion
> warn
> `GUILD_MEDIA` channels are in beta and still being actively developed. The API and other technical details are subject to change.
```

> info

> > More information about media channels and how they appear in Discord can be found in the [Media Channels Help Center Article](https://creator-support.discord.com/hc/en-us/articles/14346342766743)

Reviewer 4: ```suggestion
```

Reviewer 4: moved this up to the intro

> ### Creating Threads in Forum/Media Channels

Reviewer 4: ```suggestion
### Creating Threads in Forum and Media Channels
```

> Within a thread-only channel, threads appear as posts. They can be created using the [`POST /channels/<channel_id>/threads`](#DOCS_RESOURCES_CHANNEL/start-thread-in-forum-channel) endpoint as threads in text channels, but with [slightly different parameters](#DOCS_RESOURCES_CHANNEL/start-thread-in-forum-channel-jsonform-params). For example, when creating threads in a threads-only channel, a message is created that has the same ID as the thread. This requires you to pass parameters for both a thread *and* a message.

Reviewer 4: ```suggestion
Within thread-only channels, threads appear as posts. Threads can be created using the [`POST /channels/<channel_id>/threads`](#DOCS_RESOURCES_CHANNEL/start-thread-in-forum-channel) endpoint with [slightly different parameters](#DOCS_RESOURCES_CHANNEL/start-thread-in-forum-channel-jsonform-params) than threads in text channels. For example, when creating threads in a threads-only channel, a message is created that has the same ID as the thread which requires you to pass parameters for both a thread *and* a message.
```

> Threads in a thread-only channel have the same permissions behavior as threads in a text channel, inheriting all permissions from the parent channel, with one exception: creating a thread in a threads-only channel only requires the `SEND_MESSAGES` permission.

Reviewer 4: ```suggestion
Threads in thread-only channels have the same permissions behavior as threads in a text channel, inheriting all permissions from the parent channel, with one exception: creating a thread in a threads-only channel only requires the `SEND_MESSAGES` permission.
```

- The `topic` field is what is shown in the "Guidelines" section within the Discord client.
- The `rate_limit_per_user` field limits how frequently threads can be created. There is a new `default_thread_rate_limit_per_user` field on thread-only channels as well, which limits how often messages can be sent _in a thread_. This field is copied into `rate_limit_per_user` on the thread at creation time.
- The `available_tags` field can be set when creating or updating a channel, which determines which tags can be set on individual threads within the thread's `applied_tags` field.
- The `flags` field indicates any [channel flags](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) set for a thread-only channel. Currently only `REQUIRE_TAG` can be used, which requires that a tag from `available_tags` be specified when creating a thread in that channel.

All fields for channels, including thread-only channels, can be found in the [Channel Object](#DOCS_RESOURCES_CHANNEL/channel-object).

> ### Forum/Media Channel Thread Fields

Reviewer 4: ```suggestion
### Forum and Media Channel Thread Fields
```

A thread can be pinned within a thread-only, which will be represented as the [`PINNED` flag](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) in the `flags` field within a thread-only channel. A thread that is pinned will have the `(1 << 1)` flag set, and archiving that thread will unset the flag. A pinned thread will *not* auto-archive.

The `message_count` and `total_message_sent` fields on threads in thread-only channels will increment on `MESSAGE_CREATE` events, and decrement on `MESSAGE_DELETE` and `MESSAGE_DELETE_BULK` events. There will be no specific `CHANNEL_UPDATE` event that notifies your app of changes to those fields—instead, apps should update those values when receiving corresponding events.

All fields for threads in thread-only channels can be found in the [channel resources documentation](#DOCS_RESOURCES_CHANNEL/start-thread-in-forum-channel-jsonform-params).

> ### Forum/Media Channel Formatting

Reviewer 4: ```suggestion
### Forum and Media Channel Formatting
```

> In thread-only channels, the first message in a thread and the channel topic can both contain markdown for bulleted lists and headings (unlike text channels).

Reviewer 5: Isn't this outdated anyway? I believe you can use lists and heading anywhere now.

This is probably more accurate:

```suggestion
In thread-only channels, the channel topic can contain markdown for bulleted lists and headings (unlike text channels).
```

Reviewer 6: yes it's anywhere usable

Author: We're going to update this in a new PR after this PR gets merged in, thank you for catching this 🙌
